packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/FlipbookViewer/FlipbookViewer.js

```diff
@@ -59,6 +37,18 @@ const FlipbookViewer = ({
     }
   }, [])
 
+  const imageLocationUrl = subject?.locations[currentFrame]?.url
+
+  const { img, error, loading, subjectImage } = useSubjectImage({
+    src: imageLocationUrl,
```

Comment:
Changing this from `defaultFrameLocation` to `imageLocationUrl` means `useSubjectImage()` hook is called on every frame because `imageLocationUrl` is determined by `currentFrame` rather than `defaultFrame`. This is undesired behavior because it directly calls `onSubjectReady()` every time a volunteer flips to a new frame. `onSubjectReady()` resets the rotation between frames and pushes new metadata to the classification.

Regarding `rotation`, we want volunteers to be able to rotate the subject and 'play' the flipbook with a subject rotated. `rotation` should not be reset in between every frame.

I noticed the metadata bug because earlier this year it was introduced by https://github.com/zooniverse/front-end-monorepo/pull/5754 and fixed by https://github.com/zooniverse/front-end-monorepo/pull/5952. On this branch, when I used my flipbook test project (listed in the FEM [active projects](https://docs.google.com/spreadsheets/d/1X9345vQ_xNX6BJf9_xwVCQCjxjbWWErGmV7Xe8WpHZo/edit?gid=223454926#gid=223454926) for reference), this is what a submitted classification looks like when I flipped through 12 frames. Huge metadata causes classification exports to hang.
<img width="458" alt="Screenshot 2024-11-18 at 3 32 35 PM" src="https://github.com/user-attachments/assets/159e1f1d-0511-451d-9176-f30d41c5941d">
___
Questions - Is the change from `defaultFrameLocation` to `imageLocationUrl` intentional and required for the new SingleImageViewer with VisXZoom? How can each viewer ensure `onSubjectReady()` is called only once per subject, and not per frame? I didn't look into ImageAndTextViewer, MultiFrameViewer, and SeparateFramesViewer in detail yet, but I want to include those in this discussion too.

Reply:
> Questions - Is the change from defaultFrameLocation to imageLocationUrl intentional and required for the new SingleImageViewer with VisXZoom?

I don't think it's required. I'll try refactoring so `onSubjectReady` (`onReady` passed from `SubjectViewer`) is only called once per subject and report back.